packages/mobx-state-tree/src/types/complex-types/model.ts

```diff
@@ -532,10 +532,23 @@ export function compose<T1, S1, A1, T2, S2, A2, T3, S3, A3>(
     t2: IModelType<T2, S2>,
     t3?: IModelType<T3, S3>
 ): IModelType<T1 & T2 & T3, S1 & S2 & S3> // ...and so forth...
+export function compose(
+    t1: IModelType<any, any>,
+    t2: IModelType<any, any>,
+    ...types: Array<IModelType<any, any>>
```

Comment:
I don't think it is useful to add a variadic argument here, as it looses all typings. Can you instead follow the pattern of the first overload, but then we with more generic arguments. (See for example how that is done for unions or enumerations)

Reply:
@mweststrate, thanks for the review I will have a look